Introduce deployment and multitenancy (#25)
SAP-samples/cloud-cap-samples-java: README.md

2.  Edit your Run Configuration via **Run** > **Run Configurations...** and select enter the **Profile** `sqlite` on tab **Spring** and click **Run**.

## Deploy to SAP Cloud Platform

CAP Java applications can be deployed to the SAP Cloud Platform either in single tenant or in multitenancy mode (see [Multitenancy in CAP Java](https://cap.cloud.sap/docs/java/multitenancy) for more information.

Prerequisites:
- Install the [Cloud MTA Build Tool](https://sap.github.io/cloud-mta-build-tool/): `npm install -g mbt
- Install the [Cloud Foundry Command Line Interface](https://docs.cloudfoundry.org/cf-cli/install-go-cli.html)
- Get a Cloud Platform account to deploy the services and applications

Deploy as Single Tenant Application
- Rename `mta-st.yaml` to `mta.yaml`
- Run `mbt build`
- Run `cf login`
- Run `cf deploy mta_archives/bookshop-java-public_1.0.0.mtar`

Deploy as Multitenant Application:

> - Create a HANA Cloud Instance in your Cloud Platform space

Reviewer: ```suggestion
- Create a HANA Cloud Instance in your SAP Cloud Platform space
```

> - Run `mbt build`

Reviewer: What happened to renaming mta-mt.yaml to mta-yaml? Is that not required any more?

2.  Edit your Run Configuration via **Run** > **Run Configurations...** and select enter the **Profile** `sqlite` on tab **Spring** and click **Run**.

## Deploy to SAP Cloud Platform

> When deploying the application to the SAP Cloud Platform, it can be deployed as a single tenant application or as a multitenant application.

Reviewer: ```suggestion
CAP Java applications can be deployed to the SAP Cloud Platform either in single tenant or in multitenancy mode (see [Multitenancy in CAP Java](https://cap.cloud.sap/docs/java/multitenancy) for more information.
```

Follow the prequisites and the step by step guide for the single or multitenant scenario in order to successfully deploy the application to the SAP Cloud Platform.

Prerequisites:

> - Install the "MultiApps Archive Builder": `npm install -g mbt`

Reviewer: ```suggestion
- Install the [Cloud MTA Build Tool](https://sap.github.io/cloud-mta-build-tool/): `npm install -g mbt`
```

- Install the [Cloud Foundry Command Line Interface](https://docs.cloudfoundry.org/cf-cli/install-go-cli.html)
- Get a Cloud Platform account to deploy the services and applications

Deploy as Multitenant Application:

> Create a HANA Cloud Instance in your Cloud Platform space

Reviewer: ```suggestion
- Create a HANA Cloud Instance in your Cloud Platform space
```

> - Rename `mta-mt.yaml` to `mta.yaml`

Reviewer: Can't we use mta-mt.yaml directly?

Author: Sure.

- Run `mbt build`
- Run `cf login`
- Run `cf deploy mta_archives/bookshop-java-public_1.0.0.mtar`
- Go to another subaccount in your global account, under subscriptions and subscribe to the application you deployed
- Run `cf map-route bookshop-java-public-approuter <YOUR DOMAIN> --hostname <SUBSCRIBER TENANT>-<ORG>-<SPACE>-bookshop-java-public-approuter` or create and bind the route manually

> Deploy as Single Tenant Application

Reviewer: I would put the "simple" case first...
